### PR TITLE
entire untested data reading task

### PR DESCRIPTION
**Pull request by kevingonz19**
Branch name is misleading, as we already have a branch named DataReadingTask. 
This branch does include the data reading task. I'm assuming that everything needed for these functions has already been initialized. 
There is no test file. All I'm curious about is whether or not the MemCopy() and sizeof() lines are done correctly.

### CODE REVIEW

File: Core/Src/DataReadingTask.c [hunk 1]
```diff
@@ -1,5 +1,47 @@
...
+        HAL_StatusTypeDef error;
+
+        // Send IMU data
+        error = IMU_GetMeasurements(&IMU_Data);
+        if (error != HAL_ERROR){
+            Message.id= IMU;
+            Message.length = sizeof(IMU_Data); 
+            memcpy(&Message.data.IMUData, &IMU_Data, sizeof(IMU_Data)); // copy to pass by value and not reference
+            status = Ethernet_PutInQueue(&Message);
+        }
```
Comment (manthanand) [inline Core/Src/DataReadingTask.c:30]: Add an else statement and leave a TODO comment saying that we should do error checking here

File: Core/Src/DataReadingTask.c [hunk 2]
```diff
@@ -1,5 +1,47 @@
...
+
+        // Send CANBus data
+        status = CAN_FetchMessage(&CAN_Data);
```
Comment (manthanand) [inline Core/Src/DataReadingTask.c]: keep doing this until the CAN queue is empty (keep running until we read a pdFalse

File: Core/Src/DataReadingTask.c [hunk 3]
```diff
@@ -1,5 +1,47 @@
...
+        if (status != pdFALSE) {
+            Message.id= CAN;
+            Message.length = sizeof(CAN_Data);
+            memcpy(&Message.data.CANData, &CAN_Data, sizeof(CAN_Data));
+            status = Ethernet_PutInQueue(&Message);
+        }
+               // Send GPS data
+        status = GPS_ReadData(&GPS_Data);
```
Comment (manthanand) [inline Core/Src/DataReadingTask.c]: Read GPS data first so we can use it for the time in the SD Card read writes

File: Core/Src/DataReadingTask.c [hunk 4]
```diff
@@ -1,5 +1,47 @@
...
+        if (error != HAL_ERROR){
+            Message.id= IMU;
+            Message.length = sizeof(IMU_Data); 
+            memcpy(&Message.data.IMUData, &IMU_Data, sizeof(IMU_Data)); // copy to pass by value and not reference
+            status = Ethernet_PutInQueue(&Message);
+        }
+
+        // Send CANBus data
+        status = CAN_FetchMessage(&CAN_Data);
+        if (status != pdFALSE) {
```
Comment (tiandahuang) [inline Core/Src/DataReadingTask.c]: this could be written as ```if (CAN_FetchMessage(&CAN_Data) != pdFALSE)```

File: Core/Src/DataReadingTask.c [hunk 5]
```diff
@@ -1,5 +1,47 @@
...
+            Message.id= CAN;
```
Comment (tiandahuang) [inline Core/Src/DataReadingTask.c]:
```suggestion
            Message.id = CAN;
```

File: Core/Src/DataReadingTask.c [hunk 6]
```diff
@@ -1,5 +1,47 @@
...
+            Message.length = sizeof(CAN_Data);
+            memcpy(&Message.data.CANData, &CAN_Data, sizeof(CAN_Data));
+            status = Ethernet_PutInQueue(&Message);
+        }
+               // Send GPS data
+        status = GPS_ReadData(&GPS_Data);
+        if (status != pdFALSE) {
+            Message.id= GPS;
```
Comment (tiandahuang) [inline Core/Src/DataReadingTask.c]:
```suggestion
            Message.id = GPS;
```

File: Core/Src/DataReadingTask.c [hunk 7]
```diff
@@ -1,5 +1,47 @@
...
+        GPSData_t GPS_Data;
+
+        EthernetMSG_t Message;
+        BaseType_t status;
+        HAL_StatusTypeDef error;
+
+        // Send IMU data
+        error = IMU_GetMeasurements(&IMU_Data);
+        if (error != HAL_ERROR){
+            Message.id= IMU;
```
Comment (tiandahuang) [inline Core/Src/DataReadingTask.c]:
```suggestion
            Message.id = IMU;
```

File: Core/Src/DataReadingTask.c (hunk 4 above)
Comment (manthanand) [reply Core/Src/DataReadingTask.c]: should be while if we want to collect every can message

File: Core/Src/DataReadingTask.c [hunk 8]
```diff
@@ -1,5 +1,76 @@
...
+            SDCard_data.id = GPS_SDCard;
+            memcpy(&SDCard_data.data.GPSData, &GPS_Data, sizeof(GPS_Data));
+            SDCard_PutInQueue(&SDCard_data); // TODO: Add error handling 
+            
+
+        } else {} // TODO: error handling for GPS_ReadData()
+
+        // Send IMU data and log in SD card
+        error = IMU_GetMeasurements(&IMU_Data);
+        if (error != HAL_ERROR){
```
Comment (manthanand) [inline Core/Src/DataReadingTask.c]: You don't need the extra variable error here. Try getting rid of redundant variables.

**Review by manthanand — APPROVED** [review]

**Pull request merged**